src/pip_sync_faster/main.py

```diff
@@ -1,18 +1,75 @@
+import hashlib
+import json
 import sys
 from argparse import ArgumentParser
 from importlib.metadata import version
+from os import environ
+from os.path import abspath
+from pathlib import Path
+from subprocess import CalledProcessError, run
 
 
-def hello_world():
-    return "Hello, world!"
+def get_hash(path):
+    """Return the hash of the given file."""
+    hashobj = hashlib.sha512()
 
+    with open(path, "rb") as file:
+        hashobj.update(file.read())
+
+    return hashobj.hexdigest()
+
+
+def get_hashes(paths):
+    """Return a dict mapping the given files to their hashes."""
+    return {abspath(path): get_hash(abspath(path)) for path in paths}
+
+
+def pip_sync_maybe(src_files, args=None):
```

Comment:
```suggestion
def pip_sync(src_files, args=None):
```

I don't think we need to add `_maybe` on all functions with an early return, raises an exception, could ever go wrong etc. If this is an important detail then a doc-string is probably the place.

Reply:
I've renamed this to `pip_sync_faster()` instead